pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -146,9 +146,22 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 	}
 	logger = logger.WithValues("username", userSignup.Spec.Username)
 
+	// ###################################################################################
+	//
+	// TODO Migration code - remove this after all UserSignup instances have been migrated
+	//
+	// ###################################################################################
+	if userSignup.Spec.Deactivated && !states.Deactivated(userSignup) {
+		states.SetDeactivated(userSignup, true)
+		if err := r.client.Update(context.TODO(), userSignup); err != nil {
+			return reconcile.Result{}, err
+		}
+		// Requeue the reconciliation if the UserSignup was migrated
+		return reconcile.Result{Requeue: true}, nil
```

Comment:
you don't have to force the requeue - the spec was updated, so the change will automatically trigger a new reconcile
```suggestion
		return reconcile.Result{}, nil
```